kafka-client/src/main/java/dev/responsive/kafka/internal/db/StampedKeySpec.java

```diff
@@ -58,6 +57,16 @@ public boolean retain(final Stamped<Bytes> key) {
 
   @Override
   public int compare(final Stamped<Bytes> o1, final Stamped<Bytes> o2) {
-    return ResponsiveWindowStore.compareKeys(o1, o2);
+    return compareKeys(o1, o2);
+  }
+
+  // TODO remove generic from Stamped<> and just implement Comparable
```

Comment:
by the way, we can make `Stamped<K extends Comparable>` if we did want to keep the generic (just fun java tip, I'm still in favor of removing the generic here)

Reply:
Yeah this comment is kind of mixing concerns, ultimately I'd like to remove the generic, it just so happens that this also means we can clean up some of the comparing related code